auto2.c

```diff
@@ -873,10 +873,19 @@ void auto2_read_repo_files(url_t *url)
   };
 
   for(i = 0; i < sizeof default_list / sizeof *default_list; i++) {
-    url_read_file(url, NULL, default_list[i][0], default_list[i][1], NULL, URL_FLAG_NODIGEST);
+    // be careful not to replace an existing file unless we successfully got
+    // a new version
+    str_copy(&tmp_file, new_download());
+    if(
+      !url_read_file(url, NULL, default_list[i][0], tmp_file, NULL, URL_FLAG_NODIGEST + URL_FLAG_OPTIONAL) &&
+      util_check_exist(tmp_file)
```

Comment:
Why first try to read the file and only if that fails check if it exists? Wouldn't it make more sense the other way round?